core/definitions/src/vaa.ts

```diff
@@ -90,6 +117,28 @@ export interface VAA<PL extends PayloadLiteral = PayloadLiteral>
   readonly hash: Uint8Array;
 }
 
+//We enforce distribution over union types, e.g. have
+//    ProtocolVAA<"TokenBridge", "Transfer" | "TransferWithPayload">
+//  turned into
+//    VAA<"TokenBridge-Transfer"> | VAA<"TokenBridge-TransferWithPayload">
+//  rather than
+//    VAA<"TokenBridge-Transfer" | "TokenBridge-TransferWithPayload">
+//  because while the latter is considered more idiomatic/canonical, it actually interferes with
+//  the most natural way to narrow VAAs via querying the payloadName or payloadLiteral.
+//  (Thanks for absolutely nothing, Typescript).
+//  For example, given the TokenBridge VAA union example:
+//  if (vaa.payloadName === "Transfer")
+//    typeof vaa //no narrowing - still resolves to the union type when using the latter approach
```

Comment:
can we do a type guard fn for this that works?

Reply:
Yeah, easily, though it's still natural (most of the time you'll probably want to narrow based on payloadName, but maybe sometimes also by payloadLiteral or maybe even protocolName...?)

Do you have anything in particular in mind why you would want a dedicated type guard?

Reply:
Not really, just thinking about how to get proper narrowing


Maybe useful someplace like this https://github.com/wormhole-foundation/connect-sdk/blob/develop/platforms/evm/src/protocols/tokenBridge.ts#L244-L251 

That would become `if(TokenBridge.isTransferWithPayload(vaa))...` but def not critical. 

We can always add it later